commands/util/counters.js

```diff
@@ -0,0 +1,591 @@
+                                ttw: Math.round(move.result.effectiveCombatTime / 1000),
+                                deaths: !!move.result.effectiveDeaths ? move.result.effectiveDeaths.toFixed(1) : 0,
+                                trainers: Math.ceil(move.result.estimator * 10) / 10,
+                                legacyFlag: !!move.legacyDate ? '*' : ''});
+            }
+        };
+
+        // sort by TTW, deaths desc
+        counters.sort(function(a, b) {
+            if (a.ttw < b.ttw) return -1
+            if (a.ttw > b.ttw) return 1
+            if (a.deaths < b.deaths) return -1
+            if (a.deaths > b.deaths) return 1
+            return 0
+        })
+
+        // console.log(counters)
+
+        let content = [];
+        for (let i = 0; i < counters.length; i++) {
+            if (i <= 11) {
+                content.push(`**#${(i + 1).toString()}: ${counters[i].pokemon}**: ${counters[i].fastMove}/${counters[i].chargeMove}${counters[i].legacyFlag} - ` +
+                                `${counters[i].ttw.toString()}s | ${counters[i].deaths.toString()} | ${counters[i].trainers.toString()}`)
+            }
+            if (i === 5) {
+                content.push('')  // separate the top six from the rest
+            }
+            if (i === counters.length - 1) {
+                content.push(`\n\* - *indicates legacy move*`)
+            }
+        }
+
+        let requestInfo = `**Raid Boss**: ${boss.pbName.replace(/_/g, ' ').titleCase()}\n` +
+                            `**Raid Tier**: ${tier.name}\n` +
+                            `**Attackers**: ${level.name}\n` +
+                            `**Weather Condition**: ${weather.name}\n` +
+                            `**Friendship Level**: ${friendship.name}\n` + 
+                            `**Moveset**: ${movesetName}`;
+
+        const embed = new MessageEmbed()
+            .setColor('#43B581')
+            .addField('__Name: Fast Move/Charge Move - TTW | Deaths | # Trainers__', content)
+            .addField('__Request Info__', requestInfo)
+            .setThumbnail(boss.imageURL)
+            .setFooter('Data retrieved from https://www.pokebattler.com.')
+
+        if (level.type === 'byLevel') {
+            let levelResponse = await message.channel.send(`${message.author}, here are your results:`, embed)
+            levelResponse.preserve = true;
+        } else {
+            // DM results if personal Pokebox
+            let channelResponse = await message.channel.send(`${message.author}, I sent you a DM with your results.`)
+            channelResponse.preserve = true;       
+            let dmResponse = await message.author.send(embed)
+            dmResponse.preserve = true;
+        }
+        await message.delete();
+
+        // update Pokebattler ID if needed
+        if (newId) {
+            // check if that user record exists
+            let user = await db.DB('User')
```

Comment:
This could be simplified by using a function in the `User` class and Pine's `DB.insertIfAbsent` function. You can see an example here:
https://github.com/Googer/Professor-Pine/blob/main-dev/app/user.js#L19